### PR TITLE
Add black linting package and remove autopep8

### LINKED ISSUE #1883: Add black linting package and remove autopep8 for Featuretools
- Add black linting package and remove autopep8
- Update Makefile commands
- Add a notebook cleaner
- This should be very similar to this PR:
  - https://github.com/alteryx/woodwork/pull/1164

### PR DESCRIPTION
**Pull request by thehomebrewnerd**
### Add black linting package and remove autopep8

Closes #1883

### CODE REVIEW

**Review by gsheni — DISMISSED** [review]

**Comment by rwedge** [comment]
```
module 'jinja2.utils' has no attribute 'escape'
```

**Comment by rwedge** [comment]
> ```
> module 'jinja2.utils' has no attribute 'escape'
> ```

Issue with a dependency, unrelated

File: Makefile [hunk 1]
```diff
@@ -4,17 +4,20 @@ clean:
...
 	find . -name '*~' -delete
+	find . -name '.coverage.*' -delete
 
 .PHONY: lint
 lint:
-	flake8 featuretools && isort --check-only featuretools
-	python docs/notebook_cleaner.py check-execution
+	isort --check-only featuretools
+	python docs/notebook_version_standardizer.py check-execution
+	black featuretools -t py39 --check
```
Comment (gsheni) [inline Makefile:13]: Let's just do py310 and linting checking in only that ENV.

Comment (thehomebrewnerd) [reply Makefile]: Updated to `-t py310` which required bumping the black version as `21.9b0` did not support `-t py310`

Comment (gsheni) [reply Makefile]: Thanks!

**Review by dvreed77 — APPROVED** [review]
lgtm. Makefile changes look good and the rest look like formatting.

**Pull request merged**